File: src/main/java/com/koddy/server/file/utils/converter/FileConverter.java
package com.koddy.server.file.utils.converter;

import com.koddy.server.file.domain.model.FileExtension;
import com.koddy.server.file.domain.model.RawFileData;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

public class FileConverter {
    public static RawFileData convertFile(final MultipartFile file) {
        if (file == null || file.isEmpty()) {
            return null;
        }

        try {
            final String fileName = file.getOriginalFilename();

            return new RawFileData(
                    fileName,
                    file.getContentType(),
                    FileExtension.getExtensionViaFimeName(fileName),
                    file.getInputStream()
            );
        } catch (final IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/com/koddy/server/file/utils/converter/FileConverter.kt
package com.koddy.server.file.utils.converter

import com.koddy.server.file.domain.model.FileExtension
import com.koddy.server.file.domain.model.RawFileData
import com.koddy.server.file.exception.FileException
import com.koddy.server.file.exception.FileExceptionCode.FILE_NOT_UPLOADED
import com.koddy.server.file.exception.FileExceptionCode.UNRECOGNIZABLE_FILE
import org.springframework.web.multipart.MultipartFile

object FileConverter {
    fun convertFile(file: MultipartFile?): RawFileData {
        if (file == null || file.isEmpty) {
            throw FileException(FILE_NOT_UPLOADED)
        }

        val fileName: String? = file.originalFilename

        return RawFileData(
            fileName = fileName ?: throw FileException(UNRECOGNIZABLE_FILE),
            contentType = file.contentType ?: throw FileException(UNRECOGNIZABLE_FILE),
            extension = FileExtension.from(fileName),
            content = file.inputStream,
        )
    }
}
